instrumentation/opentelemetry-annotations-1.0/testing/src/main/java/io/opentelemetry/javaagent/instrumentation/otelannotations/AbstractTraced.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.otelannotations;
+
+public abstract class AbstractTraced<T extends U, U> {
+
+  protected static final String SUCCESS_VALUE = "Value";
+
+  protected static final IllegalArgumentException FAILURE = new IllegalArgumentException("Boom");
+
+  protected AbstractTraced() {
+    if (!getClass().getSimpleName().equals("Traced")) {
+      throw new IllegalStateException("Subclasses of AbstractTraced must be named Traced");
```

Comment:
I could instead use `getClass` in the assertions themselves but this made the assertions much nicer